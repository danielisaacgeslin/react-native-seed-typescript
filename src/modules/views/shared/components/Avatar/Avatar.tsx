import React from 'react';
import { Image } from 'react-native';

import Style from './styles';

export interface IAvatarProps {
  avatar: string;
}

export default class Avatar extends React.PureComponent<IAvatarProps> {
  public render() {
    const { avatar } = this.props;
    return <Image style={Style.avatar} source={{ uri: avatar }} />;
  }
}
