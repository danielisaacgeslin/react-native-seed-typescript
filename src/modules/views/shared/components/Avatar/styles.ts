import { StyleSheet } from 'react-native';

import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  avatar: {
    borderRadius: STYLE.BORDER.GLOBAL_RADIUS,
    height: 36,
    marginRight: 10,
    width: 36
  }
});
