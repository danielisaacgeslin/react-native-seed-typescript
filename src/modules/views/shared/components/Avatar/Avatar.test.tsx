import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

describe('Avatar', () => {
  let props;
  beforeEach(() => {
    props = {
      avatar: 'https://avatar.jpg'
    };
  });
  it('should render', () => {
    const rendered = renderer.create(<Avatar {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
