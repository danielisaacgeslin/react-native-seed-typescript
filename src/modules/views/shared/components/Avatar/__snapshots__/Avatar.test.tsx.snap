// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `
<Image
  source={
    Object {
      "uri": "https://avatar.jpg",
    }
  }
  style={
    Object {
      "borderRadius": 4,
      "height": 36,
      "marginRight": 10,
      "width": 36,
    }
  }
/>
`;
